### PR TITLE
🛡️ Sentry: [test coverage improvement]

### PR DESCRIPTION
**Pull request by madmax983**
* 🎯 Target: `src/trajectory/mod.rs` (`TokenUsage` and `extra_is_empty` logic).
* 💣 Risk: Token usage calculations and extra fields logic could break leading to incorrect metric reporting and trajectory serialization.
* 🧪 Strategy: Added table-driven tests checking various token values and edge cases (like saturation behavior).
* 🔬 Verification: Run `cargo test --manifest-path Cargo.toml --lib trajectory`

---
*PR created automatically by Jules for task [198706790524585661](https://jules.google.com/task/198706790524585661) started by @madmax983*

### CODE REVIEW

File: src/trajectory/mod.rs [hunk 1]
```diff
@@ -307,4 +307,72 @@ mod tests {
...
+        assert!(!extra_is_empty(&extra));
+
+        extra.actions = None;
+        extra.cost = Some(1.0);
+        assert!(!extra_is_empty(&extra));
+
+        extra.cost = None;
+        extra.other.insert("key".into(), serde_json::json!("val"));
+        assert!(!extra_is_empty(&extra));
+    }
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/trajectory/mod.rs:385]: > medium
> 
> The test `extra_is_empty_checks_all_fields` claims to check all fields but omits `response` and `timestamp`. To ensure the test is comprehensive and covers all fields defined in `MessageExtra`, these fields should be included in the test logic.
> 
> ```rust
>     #[test]
>     fn extra_is_empty_checks_all_fields() {
>         let mut extra = MessageExtra::default();
>         assert!(extra_is_empty(&extra));
> 
>         extra.actions = Some(vec!["action".into()]);
>         assert!(!extra_is_empty(&extra));
> 
>         extra.actions = None;
>         extra.cost = Some(1.0);
>         assert!(!extra_is_empty(&extra));
> 
>         extra.cost = None;
>         extra.response = Some(serde_json::json!("resp"));
>         assert!(!extra_is_empty(&extra));
> 
>         extra.response = None;
>         extra.timestamp = Some("2024-01-01".into());
>         assert!(!extra_is_empty(&extra));
> 
>         extra.timestamp = None;
>         extra.other.insert("key".into(), serde_json::json!("val"));
>         assert!(!extra_is_empty(&extra));
>     }
> ```

@jules
